package.json

```diff
@@ -1,17 +1,22 @@
 {
   "name": "@catchandrelease/arbor",
-  "version": "0.97.9",
+  "version": "0.98.0",
   "description": "React component library for Catch&Release",
   "main": "dist/index.js",
+  "types": "dist/index.d.ts",
   "scripts": {
     "test": "TZ=UTC jest",
     "storybook": "start-storybook -p 6006",
     "build-storybook": "build-storybook",
     "lint": "yarn eslint && stylelint src",
     "eslint": "eslint src stories .storybook",
-    "test:ci": "yarn test && yarn lint",
+    "test:ci": "yarn typecheck && yarn test && yarn lint",
     "deploy-storybook": "yarn storybook-to-ghpages",
-    "build": "rm -rf dist/* && babel src -d dist --ignore '**/*.test.js'",
+    "build": "yarn typecheck && yarn build:clean && yarn build:ts && yarn build:js",
+    "build:clean": "rm -rf dist/*",
+    "build:js": "babel src -d dist --extensions \".js,.ts,.tsx\" --ignore '**/*.test.js'",
+    "build:ts": "tsc --emitDeclarationOnly --noEmit false",
+    "typecheck": "tsc",
```

Comment:
I broke things out a bit here, but our normal workflows will still just use `build`/`test`/`test:ci` as normal.  `yarn typecheck --watch` can be handy when working through type errors, but typecheck runs as part of `test:ci` and `build` too as you can see.  (It's included in `build` for convenience so as to avoid blowing away `dist` if typechecking fails.)